test/apps/projection/cpu-line-layer.ts

```diff
@@ -0,0 +1,97 @@
+import {Layer, LayerContext, Color, UpdateParameters} from '@deck.gl/core';
+
+type CPULineLayerProps<DataT = unknown> = {
+  data: DataT[];
+  getStartPosition: (d: DataT) => number[];
+  getEndPosition: (d: DataT) => number[];
+  width?: number;
+  color?: string;
+};
+
+const defaultProps = {
+  width: 1,
+  color: 'black'
+};
+
+export class CPULineLayer<DataT> extends Layer<Required<CPULineLayerProps<DataT>>> {
+  static layerName = 'CPULineLayer';
+  static defaultProps = defaultProps;
+
+  state!: {
+    container: Element;
+  };
+
+  initializeState({device}: LayerContext) {
+    const canvas = device.canvasContext?.canvas as HTMLCanvasElement;
+    if (canvas) {
+      const container = appendSVGElement(canvas.parentElement!, 'svg');
```

Comment:
Interesting use of a deck layer.
Is it correct to say this renders a 2D grid with SVG over the canvas? I'm assuming this only works since pitch is 0?